### PR TITLE
bump go version to 1.21

### PR DESCRIPTION
**Pull request by pkbhowmick**

### CODE REVIEW

**Comment by pkbhowmick** [comment]
https://github.com/golangci/golangci-lint seems to be a blocker in go version update to 1.20

**Comment by sttts** [comment]
@pkbhowmick can you fix the lint error?

**Comment by pkbhowmick** [comment]
> @pkbhowmick can you fix the lint error?

Yeah, looking into it. It seems the linter is not compatible with 1.21

**Comment by sttts** [comment]
> Yeah, looking into it. It seems the linter is not compatible with 1.21

@pkbhowmick compare https://github.com/kcp-dev/kcp/commit/492f3e2d1b090881fe210ba9f9c1cc42cf29b250

**Comment by sttts** [comment]
https://github.com/kube-bind/kube-bind/blob/main/hack/tools.go needs to be changed, compare https://github.com/kcp-dev/kcp/commit/492f3e2d1b090881fe210ba9f9c1cc42cf29b250.

**Comment by pkbhowmick** [comment]
> [`main`/hack/tools.go](https://github.com/kube-bind/kube-bind/blob/main/hack/tools.go?rgh-link-date=2023-10-23T14%3A21%3A17Z) needs to be changed, compare [kcp-dev/kcp@492f3e2](https://github.com/kcp-dev/kcp/commit/492f3e2d1b090881fe210ba9f9c1cc42cf29b250)

.golang-ci.yaml is also missed

**Comment by pkbhowmick** [comment]
error log from linter is kinda weird

File: hack/go-install.sh [hunk 1]
```diff
@@ -57,6 +57,6 @@ cd "${tmp_dir}"
 go mod init fake/mod
 
 # install the golang module specified as the first argument
-go install -tags tools "${1}@${3}"
+go install -tags kcptools "${1}@${3}"
```
Comment (sttts) [inline hack/go-install.sh:60]: bindtools

**Comment by sttts** [comment]
I had the same strange error in kcp. Some of the changes in that linked PR fixed it :)

**Comment by sttts** [comment]
Replaced by 1.23 update.

**Pull request closed without merging**